implementations/rust/ockam/ockam_api/src/nodes/service.rs

```diff
@@ -642,6 +660,25 @@ impl NodeManagerWorker {
             (Post, ["node", "outlet"]) => self.create_outlet(req, dec).await?.to_vec()?,
             (Delete, ["node", "portal"]) => todo!(),
 
+            (Post, ["policy", resource, action]) => {
+                let this = self.node_manager.read().await;
+                this.add_policy(resource, action, req, dec)
+                    .await?
+                    .to_vec()?
+            }
+            (Get, ["policy", resource, action]) => {
+                let this = self.node_manager.read().await;
+                match this.get_policy(req.id(), resource, action).await? {
+                    Either::Left(res) => {
+                        let err = Error::new(req.path())
+                            .with_method(Method::Get)
+                            .with_message("policy not found");
+                        res.body(err).to_vec()?
+                    }
+                    Either::Right(res) => res.to_vec()?,
```

Comment:
What's the benefit of using `Either` here? I see a couple of drawbacks:

1. If you are converting the `Left` variant into an error, wouldn't it be easier to create the error within `get_policy`? 
2. Also, if the `get_policy` method evolves so it returns multiple error kinds, we would have to handle them here to adapt the error message instead of just returning the error as is, right?

Reply:
Thanks, I moved the error creation into `NodeManager::get_policy`.